app/policies/case/sar_policy.rb

```diff
@@ -7,34 +7,42 @@ def initialize(user, scope)
       @scope = scope
     end
 
+    def correspondence_type
+      CorrespondenceType.sar
+    end 
+
     def resolve
-      if @user.manager?
-        @scope
-      else
+      if @user.permitted_correspondence_types.include? correspondence_type
         scopes = []
+        if @user.manager?
+          scopes << @scope
+        end
+
+        team_restriction_sql = "select case_id from assignments 
+                join teams_users_roles 
+                on assignments.team_id = teams_users_roles.team_id
+                where teams_users_roles.role = '%<role>s' 
+                and teams_users_roles.user_id = %<user_id>i"
 
-        # Unfortunately neither of the scopes suggested here work with with the 'or' clause
-        # as they join across tables. It's not obvious whether a user can be both
-        # a responder and an approver which would result in an 'or' join here - but
-        # the higher level tries to do an 'or' join which rules this out anyway.
         if @user.responder?
-          case_ids = Assignment.with_teams(@user.responding_teams).pluck(:case_id)
-          scopes << @scope.where(id: case_ids)
+          scopes << @scope.where(" cases.id in 
+          (#{team_restriction_sql}) " % {role: :responder.to_s, user_id: @user.id})
```

Comment:
I think here again we should try to find a way to make this more ActiveRecord friendly rather than rely on raw SQL